lib/Service/RoomFormatter.php

```diff
@@ -319,6 +320,7 @@ public function formatRoomV4(
 				$lastMentionDirect = $attendee->getLastMentionDirect();
 				$roomData['unreadMention'] = $lastMention !== 0 && $lastReadMessage < $lastMention;
 				$roomData['unreadMentionDirect'] = $lastMentionDirect !== 0 && $lastReadMessage < $lastMentionDirect;
+				$roomData['lastUnreadMentionMessage'] = ($lastMentionDirect !== 0 or $lastMentionDirect !== null) ? $this->participantService->getLastUnreadMentionMessage($lastMentionDirect) : '';
```

Comment:
Removed the OR condition and updated the statement as suggested. 

For TalkWidget, indeed the changes were missed because of not knowing what is the best way to test. With the above statement check you provided, it was easily testable and hence fixed as well with 2nd check-in.